db/migrate/20141222151001_add_host_content_view_environment.rb

```diff
@@ -0,0 +1,72 @@
+class AddHostContentViewEnvironment < ActiveRecord::Migration
+  debugger
+  class Katello::ContentViewPuppetEnvironment  < Katello::Model
+    belongs_to :environment, :class_name => "Katello::KTEnvironment",
+                             :inverse_of => :content_view_puppet_environments
+    belongs_to :content_view_version, :class_name => "Katello::ContentViewVersion",
+                                      :inverse_of => :content_view_puppet_environments
+
+    belongs_to :puppet_environment, :class_name => "Environment",
+                                    :inverse_of => :content_view_puppet_environment, :dependent => :destroy
+
+    def content_view
+      self.content_view_version.content_view
+    end
+  end
+
+  class Katello::ContentViewVersion  < Katello::Model
+    belongs_to :content_view, :class_name => "Katello::ContentView", :inverse_of => :content_view_versions
+  end
+
+  class Katello::ContentView  < Katello::Model
+    has_many :content_view_versions, :class_name => "Katello::ContentViewVersion", :dependent => :destroy
+    alias_method :versions, :content_view_versions
+  end
+
+  class Katello::KTEnvironment  < Katello::Model
+    has_many :content_view_puppet_environments, :class_name => "Katello::ContentViewPuppetEnvironment",
+                                                :foreign_key => :environment_id, :inverse_of => :environment, :dependent => :restrict
+  end
+
+  class Environment   < ActiveRecord::Base
+    has_one :content_view_puppet_environment, :class_name => "Katello::ContentViewPuppetEnvironment",
+                                              :foreign_key => :puppet_environment_id,
+                                              :dependent => :nullify, :inverse_of => :puppet_environment
+
+    def content_view
+      self.content_view_puppet_environment.try(:content_view)
+    end
+
+    def lifecycle_environment
+      self.content_view_puppet_environment.try(:environment)
+    end
+  end
+
+  def up
+    add_column :hosts, :content_view_id, :integer, :null => true
+    add_column :hosts, :lifecycle_environment_id, :integer, :null => true
+
+    add_column :hostgroups, :content_view_id, :integer, :null => true
+    add_column :hostgroups, :lifecycle_environment_id, :integer, :null => true
+
+    [Hostgroup, Host::Managed].each do |model|
+      model.find_each do |host|
+        lifecycle_environment =  host.environment.try(:lifecycle_environment)
+        content_view = host.environment.try(:content_views)
+        if lifecycle_environment && content_view
+          host.content_view = content_view
+          host.lifecycle_environment = lifecycle_environment
+          host.save!
```

Comment:
This triggers callbacks, and because I have real hosts on here it attempts to do things like reach out to the smart proxy.  I think `update_columns` would be better to avoid any callbacks/validations.